### DISCUSSION TITLE
Not able to install github-changelog-generator

### DISCUSSION THREAD

**Discussion by jeevitesh** [github-changelog-generator/github-changelog-generator#1001] [General]
Hi
I'm not able to install github-changelog-generator, I'm using windows and In stalled Ruby getting below error during installation.
jeevitesh@USL-L-Jeevitesh MINGW64 /d/jeevi/cimmesb/material/Release_notes/ruby_releaseNotes
$ gem install github_changelog_generator
ERROR:  Error installing github_changelog_generator:
        ERROR: Failed to build gem native extension.

    current directory: C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/nio4r-2.5.8/ext/nio4r
C:/Ruby30-x64/bin/ruby.exe -I C:/Ruby30-x64/lib/ruby/3.0.0 -r ./siteconf20211119-25688-jtf6hc.rb extconf.rb

current directory: C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/nio4r-2.5.8/ext/nio4r
make DESTDIR\= clean
current directory: C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/nio4r-2.5.8/ext/nio4r
make DESTDIR\=
make failedNo such file or directory - make

Gem files will remain installed in C:/Ruby30-x64/lib/ruby/gems/3.0.0/gems/nio4r-2.5.8 for inspection.
Results logged to C:/Ruby30-x64/lib/ruby/gems/3.0.0/extensions/x64-mingw32/3.0.0/nio4r-2.5.8/gem_make.out
Successfully installed rainbow-3.0.0
Successfully installed ruby2_keywords-0.0.5
Successfully installed multipart-post-2.1.1
Successfully installed faraday-rack-1.0.0
Successfully installed faraday-patron-1.0.0
Successfully installed faraday-net_http_persistent-1.2.0
Successfully installed faraday-net_http-1.0.1
Successfully installed faraday-httpclient-1.0.1
Successfully installed faraday-excon-1.1.0
Successfully installed faraday-em_synchrony-1.0.0
Successfully installed faraday-em_http-1.0.0
Successfully installed faraday-1.8.0
Successfully installed public_suffix-4.0.6
Successfully installed addressable-2.8.0
Successfully installed sawyer-0.8.2
Successfully installed octokit-4.21.0
Successfully installed multi_json-1.15.0
Successfully installed faraday-http-cache-2.2.0
Successfully installed protocol-http-0.22.5
Successfully installed protocol-hpack-1.4.2
Successfully installed protocol-http2-0.14.2
Successfully installed protocol-http1-0.14.2
Successfully installed timers-4.3.3
Temporarily enhancing PATH for MSYS/MINGW...
Building native extensions. This could take a while...

jeevitesh@USL-L-Jeevitesh MINGW64 /d/jeevi/cimmesb/material/Release_notes/ruby_releaseNotes
$ github_changelog_generator -u github_project_namespace -p github_project
bash: github_changelog_generator: command not found

After getting the above error used the below error. but still same issue exists
jeevitesh@USL-L-Jeevitesh MINGW64 /d/jeevi/cimmesb/material/Release_notes/ruby_releaseNotes
$ gem install compass --pre
Successfully installed ffi-1.15.4-x64-mingw32
Successfully installed rb-inotify-0.10.1
Successfully installed rb-fsevent-0.11.0
Successfully installed sass-listen-3.0.7
Successfully installed sass-3.5.0.pre.rc.1
Successfully installed compass-import-once-1.0.5
Successfully installed compass-core-1.1.0.alpha.3
Successfully installed chunky_png-1.4.0
    Compass is charityware. If you love it, please donate on our behalf at http://umdf.org/compass Thanks!
Successfully installed compass-1.1.0.alpha.3
Parsing documentation for ffi-1.15.4-x64-mingw32
Installing ri documentation for ffi-1.15.4-x64-mingw32
Parsing documentation for rb-inotify-0.10.1
Installing ri documentation for rb-inotify-0.10.1
Parsing documentation for rb-fsevent-0.11.0
Installing ri documentation for rb-fsevent-0.11.0
Parsing documentation for sass-listen-3.0.7
Installing ri documentation for sass-listen-3.0.7
Parsing documentation for sass-3.5.0.pre.rc.1
Installing ri documentation for sass-3.5.0.pre.rc.1
Parsing documentation for compass-import-once-1.0.5
Installing ri documentation for compass-import-once-1.0.5
Parsing documentation for compass-core-1.1.0.alpha.3
Installing ri documentation for compass-core-1.1.0.alpha.3
Parsing documentation for chunky_png-1.4.0
Installing ri documentation for chunky_png-1.4.0
Parsing documentation for compass-1.1.0.alpha.3
Installing ri documentation for compass-1.1.0.alpha.3
Done installing documentation for ffi, rb-inotify, rb-fsevent, sass-listen, sass, compass-import-once, compass-core, chunky_png, compass after 9 seconds
9 gems installed

jeevitesh@USL-L-Jeevitesh MINGW64 /d/jeevi/cimmesb/material/Release_notes/ruby_releaseNotes
$ gem install github_changelog_generator
ERROR:  Error installing github_changelog_generator:
        ERROR: Failed to build gem native extension.

**Comment by olleolleolle** [comment]
👋 Hello! In order to use Ruby on Windows, with full support for C extensions (program pieces that interact with Ruby, but are written in another programming language), you can use a Ruby installation such as the https://github.com/oneclick/rubyinstaller2 

There is a description of the installer, its usage and so on at: https://github.com/oneclick/rubyinstaller2#using-the-installer-on-a-target-system 

I hope this helps!
